### ISSUE TITLE
Saving Yenniku

### ISSUE THREAD

**Issue by KydarykCC** [chromiecraft/chromiecraft#5616]
### What client do you play on?

enUS

### Faction

Horde

### Content Phase:

40-49

### Current Behaviour

I use the Soul Gem on Yenniku to release him from his imprisonment as the quest advises. Yenniku turns friendly, but immediately runs back to his spawn point and resets back to hostile. 

https://wowgaming.altervista.org/aowow/?item=3912 - link to the quest item
https://wowgaming.altervista.org/aowow/?quest=592 - link to the quest itself

### Expected Blizzlike Behaviour

I believe the stun mechanism is what is failing to trigger, stopping him from running back to his starting point. The Soul Gem should stun him, holding him in place while he is friendly so we can turn in the Soul Gem to receive the Filled Soul Gem.

https://www.youtube.com/watch?v=vdI0PJmOmmQ - link to a youtube video, showing the expected behavior.

### Source

https://www.youtube.com/watch?v=vdI0PJmOmmQ

I'm not sure what else to put here.

### Steps to reproduce the problem

1. Head to the quest area in Stranglethorn Vale. It is north of Booty Bay, but south of the Gurubashi Arena, on the eastern coast.
2. Fight your way to Yenniku, killing other Zanzil mobs in the way.
3. Approach Yenniku, and once in range (melee range) use the Soul Gem while Yenniku is targeted.
4. Attempt to click on Yenniku who should now be friendly to turn in/progress the quest forward to the next stage.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/8224a90a99cc40ff9a7a9a3ab4d7aecc40aacd3d

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-BreakingNewsOverride](https://github.com/AnchyDev/BreakingNewsOverride)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-zone-difficulty](https://github.com/azerothcore/mod-zone-difficulty)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-event-scripts](https://github.com/55Honey/Acore_eventScripts)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by KhobarKast** [comment]
Can confirm the issue. Yenniku for some reason takes exactly 1 point of damage while running away, and I had no damage effects at all on him or retaliatory stuff on my Shaman. Happens repeatedly. That 1 point of damage takes him out of the stun, this quest is unfortunately undoable currently.

**Comment by Paradigmxd** [comment]
21.05.2023 this quest is still broken and therefore interrupts the completion of the quest line...

Very sad indeed since it gives good items for leveling...

**Comment by gchekenny** [comment]
Tested, quest works as intended
![WoWScrnShot_061623_192017](https://github.com/chromiecraft/chromiecraft/assets/65800273/613d4402-4fcc-46b2-b3be-8f4c659fc3db)

**Comment by KhobarKast** [comment]
There must have been a change somewhere along the line. Either way, I can confirm, also working for me now!

**Comment by The-GhostRider** [comment]
I confirm the bug. The issue arises when Yenniku changes faction after casting the spell of the quest item. At that point, he draws aggro from the surrounding mobs, and when he enters an evasive state, he resets.

https://github.com/chromiecraft/chromiecraft/assets/1884642/3c563272-952a-48cc-9fec-7ac81944808a


Anyway, today while I was in the game, I noticed the issue because I also completed the quest with my character, and I have already fixed the problem. I just need to submit the PR.
